tracing-appender/src/lib.rs

```diff
@@ -7,6 +7,19 @@ pub mod non_blocking;
 pub mod rolling;
 mod worker;
 
+/// A function for which accepts a struct implementing the `Write` Trait and provides a
+/// a `MakeWriter` which allows writes to happen in a non blocking manner.
+/// # Examples
+/// ``` rust,ignore
+/// use tracing_subscriber::fmt::MakeWriter;
+///
+/// let (non_blocking, _guard) = non_blocking(std::io::stdout());
+/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+/// tracing::subscriber::with_default(subscriber.finish(), || {
+///    tracing::event!(tracing::Level::INFO, "Hello");
+/// });
+///
+/// ```
```

Comment:
```suggestion
/// Creates a non-blocking, off-thread writer.
///
/// Note that the `WorkerGuard` returned by `non_blocking` _must_ be assigned to a binding that is not `_`,
/// as `_` will result in the `WorkerGuard` being dropped immediately.
///
/// # Examples
/// ```
/// use tracing_subscriber::fmt::MakeWriter;
///
/// let (non_blocking, _guard) = non_blocking(std::io::stdout());
/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
/// tracing::subscriber::with_default(subscriber.finish(), || {
///     tracing::event!(tracing::Level::INFO, "Hello");
/// });
/// ```
```